packages/storage-plus/src/snapshot_item.rs

```diff
@@ -81,26 +80,11 @@ where
         self.primary.may_load(store)
     }
 
-    // may_load_at_height reads historical data from given checkpoints.
-    // Only returns `Ok` if we have the data to be able to give the correct answer
-    // (Strategy::EveryBlock or Strategy::Selected and h is registered as checkpoint)
-    //
-    // If there is no checkpoint for that height, then we return StdError::NotFound
     pub fn may_load_at_height(&self, store: &dyn Storage, height: u64) -> StdResult<Option<T>> {
-        self.assert_checkpointed(store, height)?;
-
-        // this will look for the first snapshot of the given address >= given height
-        // If None, there is no snapshot since that time.
-        let start = Bound::inclusive(U64Key::new(height));
-        let first = self
-            .snapshots
-            .changelog
-            .range(store, Some(start), None, Order::Ascending)
-            .next();
-
-        if let Some(r) = first {
-            // if we found a match, return this last one
-            r.map(|(_, v)| v.old)
+        let snapshot = self.snapshots.may_load_at_height(store, (), height)?;
+
+        if let Some(r) = snapshot {
```

Comment:
Yes, this looks nice here, keeping the primary load in this function, and checking an option to switch.